### ISSUE TITLE
Ack handling

### ISSUE THREAD

**Issue by ltigges** [ioBroker/ioBroker.mysensors#58]
Hi all,
In my opinion, the ack flag is not processed correctly. Currently the messages are sent with ack request and the ack flag is set in the iobroker as soon as the ack confirmation arrives.
However, the IOB ack is normally set as soon as an actuator has performed an action and not as soon as the data transmission was successful. Therefore ideally the Mysensors ack should be used to retrigger the data transfer if it fails but in no way to set the IOB ack flag.

**Comment by Apollon77** [comment]
This depends on the device. IF there is a reponse of the device about the change nut no new values then a successfull sending is the only way to set the ack flag ...

**Comment by ltigges** [comment]
As I understand it, the IOB ack should only be set when the command has been completed. With a relay the current implementation works but for example with a roller shutter that is controlled by V_PERCENTAGE , the iob ack is already set although the roller shutter is still running.

**Comment by Apollon77** [comment]
> As I understand it, the IOB ack should only be set when the command has been completed.

Yes and no ... formally it is about "the device acknowledged the receiving" - what that in the end means is partly with the device, but when the device has active data reporting no adapter will set "ack=true" itself.

mysensors might be special here because I do not know if the device always report the new value back ... so I can only assume (without looking into it) that the "direct ack" is a workaround in this case ...
